homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -242,7 +242,7 @@ def update_data():
         for sub_device in gateway.gateway_device.devices.values():
             try:
                 sub_device.update()
-            except GatewayException as ex:
+            except (GatewayException, TypeError) as ex:
                 _LOGGER.error("Got exception while fetching the state: %s", ex)
```

Comment:
We schould probablly log the error only once.
Could you include a:
`if data.get(sub_device.sid, {}).get(ATTR_AVAILABLE):`